pkg/migrations/sql/idempotent/base.sql

```diff
@@ -614,15 +634,26 @@ GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.delete_series_catalog_row(name, bigint[
 ---------------------------------------------------
 
 CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.get_metric_table_name_if_exists(
-        metric_name text)
-    RETURNS TABLE (id int, table_name name)
+        schema text, metric_name text)
+    RETURNS TABLE (id int, table_name name, table_schema name, series_table name, is_view boolean)
 AS $func$
-   SELECT id, table_name::name
+   SELECT id, table_name::name, table_schema::name, series_table::name, is_view
+   FROM SCHEMA_CATALOG.metric m
+   WHERE m.table_schema = get_metric_table_name_if_exists.schema
+   AND m.metric_name = get_metric_table_name_if_exists.metric_name
+   UNION ALL
+   SELECT id, table_name::name, table_schema::name, series_table::name, is_view
+   FROM SCHEMA_CATALOG.metric m
+   WHERE m.table_schema = 'SCHEMA_DATA'
+   AND m.metric_name = get_metric_table_name_if_exists.metric_name
+   UNION ALL
+   SELECT id, table_name::name, table_schema::name, series_table::name, is_view
```

Comment:
Everythig looks good except shouldn't the last (3rd case) error if there are more than 2 schemas with same name?

Reply:
I think the logic we discussed was:
if schema is specified use the exact match
else {
   if there is a metric in the data schema return that
   else {
      if there is only one metric across all schemas, return that
       else throw error
    }
}